src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java

```diff
@@ -85,6 +88,29 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             .build();
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("Submit petition about to submit callback invoked");
+
+        final CaseData caseData = details.getData();
+
+        if (isNotAccepted(caseData.getStatementOfTruth())
```

Comment:
I wonder if these would be better as a method on the model, e.g. `caseData.hasStatementOfTruth()`

Reply:
Yes, agreed.